src/textbooks/textbook-form/TextbookForm.jsx

```diff
@@ -0,0 +1,192 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import * as Yup from 'yup';
+import { FieldArray, Formik } from 'formik';
+import {
+  PictureAsPdf as PdfIcon,
+  Add as AddIcon,
+  DeleteOutline as DeleteIcon,
+  Upload as UploadIcon,
+} from '@edx/paragon/icons';
+import {
+  ActionRow,
+  Button,
+  Form,
+  Icon,
+  IconButtonWithTooltip,
+  useToggle,
+} from '@edx/paragon';
+
+import FormikControl from '../../generic/FormikControl';
+import PromptIfDirty from '../../generic/PromptIfDirty';
+import ModalDropzone from '../../generic/modal-dropzone/ModalDropzone';
+import { useModel } from '../../generic/model-store';
+import messages from './messages';
+
+const TextbookForm = ({
+  closeTextbookForm,
+  initialFormValues,
+  onSubmit,
+  onSavingStatus,
+  courseId,
+}) => {
+  const intl = useIntl();
+
+  const courseDetail = useModel('courseDetails', courseId);
+  const courseTitle = courseDetail ? courseDetail?.name : '';
+
+  const [currentTextbookIndex, setCurrentTextbookIndex] = useState(0);
+  const [isUploadModalOpen, openUploadModal, closeUploadModal] = useToggle(false);
+
+  const textbookFormValidationSchema = Yup.object().shape({
+    tab_title: Yup.string().required(intl.formatMessage(messages.tabTitleValidationText)).max(255),
+    chapters: Yup.array().of(
+      Yup.object({
+        title: Yup.string().required((intl.formatMessage(messages.chapterTitleValidationText))).max(255),
+        url: Yup.string().required(intl.formatMessage(messages.chapterUrlValidationText)).max(255),
+      }),
+    ).min(1),
+  });
+
+  return (
+    <div className="textbook-form">
+      <Formik
+        initialValues={initialFormValues}
+        validationSchema={textbookFormValidationSchema}
+        onSubmit={onSubmit}
+        validateOnBlur
+        validateOnMount
+      >
+        {({
+          values, handleSubmit, isValid, dirty, setFieldValue,
+        }) => (
+          <>
+            <Form.Group size="sm" className="form-field">
```

Comment:
[important]: Please try to fix the error in the console "Failed prop type: Invalid prop `isInvalid` of type `string` supplied to `ForwardRef`, expected `boolean`." this error appears when `Form.Control.Feedback` is displayed when validating inputs

Reply:
![Screenshot 2024-02-11 at 14 53 55](https://github.com/raccoongang/frontend-app-course-authoring/assets/138868841/215b60df-1f56-453a-9294-061f82797097)

It can be fixed in FormikControl generic